src/DynamoSandbox/DynamoCoreSetup.cs

```diff
@@ -76,7 +76,19 @@ public void RunApplication(Application app)
                 Dynamo.Applications.StartupUtils.ASMPreloadFailure -= ASMPreloadFailureHandler;
 
             }
+            catch(FileLoadException e)
+            {
+                DynamoModel.IsCrashing = true;
+                // If the exception is having HRESULT of 0x80131515, then we need to instruct the user to "unblock" the downloaded DLL.
+                if (e.HResult == unchecked((int)0x80131515))
+                {
+                    var failureMessage = string.Format(Dynamo.Properties.Resources.LibraryPreloadFailureForBlockedAssembly, e.Message);
+                    MessageBox.Show(failureMessage, "DynamoSandbox", MessageBoxButton.OK, MessageBoxImage.Warning);
 
+                    Debug.WriteLine(e.Message);
+                    Debug.WriteLine(e.StackTrace);
+                }
```

Comment:
All such exceptions thrown during preloading libraries that are blocked finally end up here, the message box is presented and Dynamo shuts down silently.

Reply:
I am worried (in general) that someone might come along and write a try catch that will silence the blocked dll exception before it gets a chance to bubble up to here.

Reply:
I have tried to make it a point not to catch other types of exceptions and rethrow them where necessary while working on this task and I hope that developers are rigorous about this. Do you know of any best practices around this? I can add more comments where they might make sense.

Reply:
What about having a single event handler "assemblyblockedhabdler(message, etc" that ca log and also end the process.
You could invoke this event in every try catch where the blocking failure is identified. This would ensure nobody has a chance to continue executing code after the error ia identified.